vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/editor/EditorRenderer.java

```diff
@@ -149,46 +149,46 @@ public void refreshData(T item) {
-         * don't set the innerHTML of the <template> elements in advance, only
-         * before the client response.
-         */
+        // Run editor renderer setup
         runBeforeClientResponse(container,
-                context -> setupEditorTemplate(container, contentTemplate,
-                        context));
+                context -> setupEditorRenderer(container, context));
+
+        // Also run editor renderer setup whenever the component gets attached
+        container.addAttachListener(event -> {
+            runBeforeClientResponse(container,
+                    context -> setupEditorRenderer(container, context));
+        });
 
         return new EditorRendering(contentTemplate);
     }
 
-    private void setupEditorTemplate(Element container, Element contentTemplate,
+    private void setupEditorRenderer(Element container,
             ExecutionContext context) {
-        final String originalTemplate;
-        final Element attachedTemplate;
-
-        // the template can be null if the original renderer didn't use any
-        // template, such as the ColumnPathRenderer
-        if (contentTemplate == null) {
-            originalTemplate = "[[item." + columnInternalId + "]]";
-            attachedTemplate = new Element("template");
-            container.appendChild(attachedTemplate);
-        } else {
-            attachedTemplate = contentTemplate;
-            originalTemplate = contentTemplate.getProperty("innerHTML");
-        }
         String appId = context.getUI().getInternals().getAppId();
         String editorTemplate = String.format(
-                "<flow-component-renderer appid='%s' nodeid='[[item._%s_editor]]'></flow-component-renderer>",
+                "<flow-component-renderer appid='%s' nodeid='${model.item._%s_editor}'></flow-component-renderer>",
                 appId, columnInternalId);
 
-        attachedTemplate.setProperty("innerHTML", String.format(
         //@formatter:off
-        "<template is='dom-if' if='[[item._editing]]' restamp>%s</template>" +
-        "<template is='dom-if' if='[[!item._editing]]' restamp>%s</template>",
+        container.executeJs("const originalRender = this.renderer;" +
+            // Patch the container's renderer function to handle the editor
+            "this.renderer = (root, container, model) => {" +
+                "const editingChanged = root.__editing !== model.item._editing;" +
+                "root.__editing = model.item._editing;" +
+
+                // If the editing state changed, the root needs to be cleared
+                "if (editingChanged) {" +
+                    "delete root._$litPart$; root.innerHTML = ''" +
+                "}" +
+
+                // If editing, render the editor, otherwise use the original renderer
+                "if (root.__editing) { root.innerHTML = `" + editorTemplate + "` }" +
+                "else if (!originalRender) { root.textContent = model.item." + columnInternalId + " }" +
```

Comment:
A fallback mechanism for columns without a `renderer` function. Previously [here](https://github.com/vaadin/flow-components/blob/0aa2a97409b050789a9ae21f216e33c6530e7ea2/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/editor/EditorRenderer.java#L172-L174)